io/zenoh-transport/src/unicast/manager.rs

```diff
@@ -473,28 +479,31 @@ impl TransportManager {
         super::establishment::open::open_link(&link, self).await
     }
 
-    pub async fn get_transport_unicast(&self, peer: &ZenohId) -> Option<TransportUnicast> {
-        zasynclock!(self.state.unicast.transports)
-            .get(peer)
-            .map(|t| t.into())
+    pub fn get_transport_unicast(&self, peer: &ZenohId) -> Option<TransportUnicast> {
```

Comment:
why turn it into a sync function if the inner code is still async? if it's for caller's benefit, shouldn't caller be the one doing task::block_on?

Reply:
To simply avoid changes in the routing code. There are few changes going on there and I'd like to minimise the impact external to the transport.

Reply:
I've moved the functions to `async` and in the caller a `block_on` has been introduced.